### DISCUSSION TITLE
Can't understand the whitelist options work

### DISCUSSION THREAD

**Discussion by mikeluigijean** [nlpsuge/gnome-shell-extension-another-window-session-manager#81] [Q&A]
what are the switcher buttons "close windows" and "Log Out, Reboot, Power Off" for?

**Comment by nlpsuge** [comment]
I suppose you mean the switcher buttons "Close windows" and "Log Out, Reboot, Power Off"  in `Whitelist` under `Close windows` in Settings.

Apps in the whitelist will be closed even they each have multiple windows and those windows are closed one by one.

"Close windows" and "Log Out, Reboot, Power Off" indicate when this happens specifically. 

For instance, I add `Extensions` to the `Whitelist` and enable all switcher buttons:
![image](https://github.com/nlpsuge/gnome-shell-extension-another-window-session-manager/assets/2271720/ac930e50-8bbc-4829-b497-482e2177171e)

When I close windows via:
![image](https://github.com/nlpsuge/gnome-shell-extension-another-window-session-manager/assets/2271720/17e9870f-4b65-4ed6-aff0-fae00d72e40d)

Or via:
![image](https://github.com/nlpsuge/gnome-shell-extension-another-window-session-manager/assets/2271720/dd6632ac-1fe3-4ef9-b5ca-74a821f13553)

AWSM will close all windows of `Extensions`.

**Reply by KevinBowen0** [reply]
But I don't understand the significance of having AWSM close the windows vs just letting them get killed on logout. What is the advantage of one way vs the other, and what does it have to do with session management?

This project could really use some documentation, or at least an informal tutorial, because it's not at all self-explanatory. I'm just trying to set up the most basic possible use case - just save and restore all my windows on logout/login. It seems like that's a simple enough use case that it should take no more than checking a single box, but after multiple days of trial and error I still haven't managed to make that work. What am I missing?

**Reply by nlpsuge** [reply]
> But I don't understand the significance of having AWSM close the windows vs just letting them get killed on logout.

Note that if we allow apps to be killed on logout, we may lost the important data that is used to restore the windows. 

For instance, consider Chrome browser. Open multiple windows and enable `On start-up -> Continue where you left off`, if we just kill the Chrome on logout, the windows may not reopen. In the worst-case scenario, you can't even find the tabs in the `History -> Recent tabs`. 

The above is the reason why I can't kill apps on logout, I prefer to close them gracefully or let users close them manually.


> I'm just trying to set up the most basic possible use case - just save and restore all my windows on logout/login.

I'm really sorry for the poor documentation. For this feature to work, you need to enable this option:
![image](https://github.com/nlpsuge/gnome-shell-extension-another-window-session-manager/assets/2271720/aebf97ad-fb97-4541-9184-29ba0a3bff96)

**Reply by KevinBowen0** [reply]
When I do that though, the only app that seems to get restored (mostly) properly is Firefox (which has always been pretty good about automatically saving and restoring its sessions even if killed. Its windows get places on the correct workspace MOST of the time, but sometimes not (and never with the correct tiling). Adding a close by rule, and/or a whitelist for it, doesn't seem to change anything - it still comes up in the warning dialog on logout (even if whitelisted). 

And every app other than firefox doesn't seem to work at all - it will restore only a single of the app's windows.

What am I doing wrong?

**Reply by duchess208** [reply]
How do I name the Whitelist apps? I'm on Ubuntu and I tried typing Firefox Web Browser and it still errored during restart. I tried to choose the application, but it didn't do anything either.
